src/main/scala/format/pgn/Parser.scala

```diff
@@ -98,296 +63,200 @@ object Parser {
+          }
       }
 
-    def strMoves: Parser[(InitialPosition, List[StrMove], Option[String])] =
-      as("moves") {
-        (commentary *) ~ (strMove *) ~ (result ?) ~ (commentary *) ^^ { case coms ~ sans ~ res ~ _ =>
-          (InitialPosition(cleanComments(coms)), sans, res)
-        }
+    def moves(str: String): Validated[String, Sans] =
+      strMove.rep.map(xs => Sans(xs.toList)).parse(str) match {
+        case Right((_, str)) =>
+          valid(str)
+        case Left(err) => invalid(showExpectations("Cannot parse moves", str, err))
       }
 
-    val moveRegex =
-      """(?:(?:0\-0(?:\-0|)[\+\#]?)|[PQKRBNOoa-h@][QKRBNa-h1-8xOo\-=\+\#\@]{1,7})[\?!□]{0,2}""".r
-
-    def forbidNullMove: Parser[Unit] =
-      as("forbidNullMove") {
-        val nullMove = "--" | "Z0" | "null" | "pass" | "@@@@"
-        guard(nullMove) ~> err("Lichess does not support null moves") | success(())
+    def move(str: String): Validated[String, San] =
+      strMove.parse(str) match {
+        case Right((_, str)) =>
+          valid(str)
+        case Left(err) => invalid(showExpectations("Cannot parse move", str, err))
       }
 
-    def strMove: Parser[StrMove] =
-      as("move") {
-        ((number | commentary) *) ~ forbidNullMove ~>
-          (moveRegex ~ nagGlyphs ~ rep(commentary) ~ nagGlyphs ~ rep(variation)) <~
-          (moveExtras *) ^^ { case san ~ glyphs ~ comments ~ glyphs2 ~ variations =>
-            StrMove(san, glyphs merge glyphs2, cleanComments(comments), variations)
-          }
-      }
+    val blockCommentary: P[String] = P.until0(P.char('}')).with1.between(P.char('{'), P.char('}'))
 
-    def number: Parser[String] = """[1-9]\d*[\s\.]*""".r
+    val inlineCommentary: P[String] = P.char(';') *> P.until(R.lf)
 
-    def moveExtras: Parser[Unit] =
-      as("moveExtras") {
-        commentary.^^^(())
-      }
+    val commentary = (blockCommentary | inlineCommentary).withContext("Invalid comment") <* whitespaces
 
-    def nagGlyphs: Parser[Glyphs] =
-      as("nagGlyphs") {
-        rep(nag) ^^ { nags =>
-          Glyphs fromList nags.flatMap { Glyph.find _ }
-        }
-      }
+    val result: P[String] = P.stringIn(List("*", "1/2-1/2", "½-½", "0-1", "1-0"))
 
-    val nagGlyphsRE = Glyph.PositionAssessment.all
-      .map(_.symbol)
-      .sortBy(-_.length)
-      .map(Regex.quote(_))
-      .mkString("|")
-      .r
+    val nagGlyphsRE = P.stringIn(
+      Glyph.PositionAssessment.all
+        .map(_.symbol)
+        .sortBy(-_.length)
+    )
 
-    def nag: Parser[String] =
-      as("nag") {
-        """\$\d+""".r | nagGlyphsRE
-      }
+    val nag = (P.char('$') ~ R.digit.rep).string | nagGlyphsRE
 
-    def variation: Parser[List[StrMove]] =
-      as("variation") {
-        "(" ~> strMoves <~ ")" ^^ { case (_, sms, _) => sms }
+    val nagGlyphs: P0[Glyphs] = (nag <* whitespaces).rep0.map(nags =>
+      Glyphs fromList nags.flatMap {
+        Glyph.find
       }
+    )
+
+    val moveExtras = commentary.as(())
```

Comment:
`commentary.void` would be a bit faster (it would not do any allocations internally to capture since it can tell you don't use the result).